source/modes/GameMode.cpp

```diff
@@ -1070,63 +1035,93 @@ bool GameMode::keyPressedChat(const OIS::KeyEvent &arg)
 void GameMode::receiveChat(ChatMessage* message)
 {
     mChatMessages.emplace_back(message);
+
+    // Adds the message right away
+    CEGUI::Window* chatTextBox = getModeManager().getGui().getGuiSheet(Gui::inGameMenu)->getChild("GameChatWindow/GameChatText");
+    chatTextBox->appendText(reinterpret_cast<const CEGUI::utf8*>(message->getMessageAsString().c_str()));
+
+    // Ensure the latest text is shown
+    CEGUI::Scrollbar* scrollBar = reinterpret_cast<CEGUI::Scrollbar*>(chatTextBox->getChild("__auto_vscrollbar__"));
```

Comment:
Check again. It should be possible to static_cast a parent class to a child class. Doing that for the scrollbar works fine here.